esrally/storage/_http.py

```diff
@@ -82,101 +87,121 @@ def head(self, url: str) -> Head:
 
-def accept_ranges_from_headers(headers: CaseInsensitiveDict) -> bool | None:
-    got = headers.get("accept-ranges", "").strip()
-    if not got:
-        return None
-    return got == "bytes"
+    @classmethod
+    def _make_head(cls, url: str, headers: CaseInsensitiveDict) -> Head:
+        content_length = cls._content_length_from_headers(headers)
+        accept_ranges = cls._accept_ranges_from_headers(headers)
+        ranges, document_length = cls._content_range_from_headers(headers)
+        hashes = cls._hashes_from_headers(headers)
+        crc32c = hashes.get("crc32c")
+        return Head.create(
+            url=url,
+            content_length=content_length,
+            accept_ranges=accept_ranges,
+            ranges=ranges,
+            document_length=document_length,
+            crc32c=crc32c,
+        )
 
+    @classmethod
+    def _content_length_from_headers(cls, headers: CaseInsensitiveDict) -> int | None:
+        value = headers.get("content-length", "").strip()
+        if not value:
+            return None
+        try:
+            return int(value)
+        except ValueError:
+            raise ValueError(f"invalid content-length value: {value}") from None
 
-def content_range_from_headers(headers: CaseInsensitiveDict) -> tuple[RangeSet, int | None]:
-    content_range_text = headers.get("content-range", "").strip()
-    if not content_range_text:
-        return NO_RANGE, None
+    @classmethod
+    def _accept_ranges_from_headers(cls, headers: CaseInsensitiveDict) -> bool | None:
+        got = headers.get("accept-ranges", "").strip()
+        if not got:
+            return None
+        return got == "bytes"
 
-    if not content_range_text.startswith("bytes "):
-        raise ValueError(f"invalid content range: {content_range_text}")
+    @classmethod
+    def _content_range_from_headers(cls, headers: CaseInsensitiveDict) -> tuple[RangeSet, int | None]:
+        content_range_text = headers.get("content-range", "").strip()
+        if not content_range_text:
+            return NO_RANGE, None
 
-    if "," in content_range_text:
-        raise ValueError("multi range value is not unsupported")
+        if not content_range_text.startswith("bytes "):
+            raise ValueError(f"invalid content range: {content_range_text}")
 
-    try:
-        value, document_length_text = content_range_text[len("bytes ") :].strip().replace(" ", "").split("/", 1)
-        document_length: int | None = None
-        if document_length_text != "*":
-            document_length = int(document_length_text)
-        return rangeset(value), document_length
-    except ValueError:
-        raise ValueError(f"invalid content range in '{content_range_text}'")
+        if "," in content_range_text:
+            raise ValueError("multi range value is not unsupported")
 
+        try:
+            value, document_length_text = content_range_text[len("bytes ") :].strip().replace(" ", "").split("/", 1)
```

Comment:
`"bytes "` here feels like a magic value, you've used it twice. Mind pulling it into a variable?